drivers/at25xxx/at25xxx.c

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (C) 2019 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_at25xxx
+ * @{
+ *
+ * @file
+ * @brief       Driver for the AT25xxx family of SPI-EEPROMs.
+ *              This also includes M95xxx, 25AAxxx, 25LCxxx,
+ *              CAT25xxx & BR25Sxxx.
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ * @}
+ */
+
+#include <errno.h>
+#include <string.h>
+
+#include "at25xxx.h"
+#include "at25xxx_params.h"
+#include "byteorder.h"
+
+#ifdef USEMODULE_XTIMER
+#include "xtimer.h"
+#define POLL_DELAY_US   (1000)
+#endif
+
+#ifndef min
+#define min(a, b) ((a) > (b) ? (b) : (a))
+#endif
+
+#define PAGE_SIZE   (AT25XXX_PARAM_PAGE_SIZE)
+#define ADDR_LEN    (AT25XXX_PARAM_ADDR_LEN)
+#define ADDR_MSK    ((1UL << ADDR_LEN) - 1)
+
+#define CMD_WREN    (0x6)   /* Write Enable             */
+#define CMD_WRDI    (0x4)   /* Write Disable            */
+#define CMD_RDSR    (0x5)   /* Read Status Register     */
+#define CMD_WRSR    (0x1)   /* Write Status Register    */
+#define CMD_READ    (0x3)   /* Read from Memory Array   */
+#define CMD_WRITE   (0x2)   /* Write to Memory Array    */
+
+#define SR_WIP      (0x01)  /* Write In Progress        */
+#define SR_WEL      (0x02)  /* Write Enable Latch       */
+#define SR_BP0      (0x04)  /* Block Protect 0          */
+#define SR_BP1      (0x08)  /* Block Protect 1          */
+#define SR_SRWD     (0x80)  /* Status Register Write Disable */
```

Comment:
(Just a suggest, I'm not insisting on this:)

You could move the defines into a header, such as `drivers/at25xxx/include/at25xxx_constants.h`. Most other RIOT drivers do so and this would make the overall look and feel of the code more consistent. A side benefit would be that Doxygen documentation could be used in that case.

Reply:
I'll consider it ;)

It's only 11 constants though, so the file header alone is longer than that. I think there is also value in keeping things contained to as few files as necessary as long as the files are still small.